reframe/core/schedulers/slurm.py

```diff
@@ -386,7 +386,14 @@ def wait(self, job):
 
         intervals = itertools.cycle(settings().job_poll_intervals)
         self._update_state(job)
+        start_pending = time.time()
         while not slurm_state_completed(job.state):
+            if max_queuing_time:
+                if slurm_state_pending(job.state):
+                    if time.time() - start_pending > max_queuing_time:
+                        self.cancel(job)
+                        raise JobError('maximum queuing time exceeded')
+
```

Comment:
What I proposed yesterday was to change how wait is implemented at the `Job` level so that it could allow us to implement this in a scheduler-agnostic way. Now, you have to implement the same logic for `wait()` in all backends.

I can do a PR changing the logic and you can continue with this feature.

Reply:
I looked into the code in more detail. I don't think that this can be done. The `local` scheduler uses a completely different mentality when waiting. So passing a timeout argument to the `wait()` method is fine.